gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/DagActionStore.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.runtime.api;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Collection;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+
+public interface DagActionStore {
+  enum DagActionValue {
+    KILL,
+    RESUME
+  }
+
+  @Getter
+  @EqualsAndHashCode
+  class DagAction {
+    String flowGroup;
+    String flowName;
+    String flowExecutionId;
+    DagActionValue dagActionValue;
+    public DagAction(String flowGroup, String flowName, String flowExecutionId, DagActionValue dagActionValue) {
+      this.flowGroup = flowGroup;
+      this.flowName = flowName;
+      this.flowExecutionId = flowExecutionId;
+      this.dagActionValue = dagActionValue;
+    }
+  }
+
+
+  /**
+   * Check if an action exists in dagAction store by flow group, flow name and flow execution id.
+   * @param flowGroup flow group for the dag action
+   * @param flowName flow name for the dag action
+   * @param flowExecutionId flow execution for the dag action
+   * @throws IOException
+   */
+  boolean exists(String flowGroup, String flowName, String flowExecutionId) throws IOException, SQLException;
+
+  /**
+   * Persist the dag action in {@link DagActionStore} for durability
+   * @param flowGroup flow group for the dag action
+   * @param flowName flow name for the dag action
+   * @param flowExecutionId flow execution for the dag action
+   * @param dagActionValue the value of the dag action
+   * @throws IOException
+   */
+  void addDagAction(String flowGroup, String flowName, String flowExecutionId, DagActionValue dagActionValue) throws IOException;
+
+  /**
+   * delete the dag action from {@link DagActionStore}
+   * @param flowGroup flow group for the dag action
+   * @param flowName flow name for the dag action
+   * @param flowExecutionId flow execution for the dag action
+   * @throws IOException
+   */
+  boolean deleteDagAction(String flowGroup, String flowName, String flowExecutionId) throws IOException;
```

Comment:
not an important one, but why deleteDagAction returns a boolean while addDagAction returns nothing?

Reply:
As long as add action does not throw exception, add should succeed. But delete can be true or false as even we try to delete some entry that does not exist, it does not throw exception, in this case, we need the boolean result to determine whether we delete the entry or not.

Reply:
Ah ok, let's update the javadoc with `@returns` then.